lib/mapUtils.js

```diff
@@ -43,3 +49,196 @@ export async function getProductData(store) {
+  else if (hoursStrSplit.length > 2) {
+    // e.g. timeAndDayPairs = ['9am-8pm Tu-Sat', '9am-7pm Sun', ...]
+    const timeAndDayPairs = formattedHours.split(', ');
+
+    timeAndDayPairs.map((timeAndDayPair, _) => {
+      // e.g. timeAndDaySplit = ['9am-8pm', 'Tu-Sat']
+      const timeAndDaySplit = timeAndDayPair.split(' ');
+      // e.g. 9am-8pm
+      const time = timeAndDaySplit[0];
+      // e.g. [Tu, Sat] or [Mon]
+      const days = timeAndDaySplit[1].split('-');
+
+      // Case: Time corresponds to one day
+      if (days.length === 1) {
+        hoursPerDayDict[daysOfTheWeek[days[0]]] = time;
+      } else if (days.length === 2) {
+        const firstDay = days[0].charAt(0).toUpperCase() + days[0].slice(1);
+        const lastDay = days[1].charAt(0).toUpperCase() + days[1].slice(1);
+
+        // ['Sun', 'Mon', 'Tues', ...]
+        const indexFirstDay = daysOfTheWeekAbbrev.indexOf(firstDay);
+        const indexLastDay = daysOfTheWeekAbbrev.indexOf(lastDay);
+
+        let numDaysInInterval;
+
+        if (indexLastDay > indexFirstDay) {
+          // Case: Interval is Sun-Thurs, 0-4
+          numDaysInInterval = indexLastDay - indexFirstDay + 1;
+        } else {
+          // Case: Interval is Thurs-Sun, 4-0
+          numDaysInInterval = indexLastDay - indexFirstDay + 8;
+        }
+
+        [...Array(numDaysInInterval).keys()].map((i, _) => {
+          const dayIndex = (indexFirstDay + i) % 7;
+          hoursPerDayDict[daysOfTheWeekFull[dayIndex]] = time;
+        });
+      } else {
+        // Shouldn't ever have a case with more than two days in an interval.
+        console.log(
+          '[mapUtils] constructHoursDict: Incorrect store hours formatting in Airtable.'
+        );
+        return null;
+      }
+    });
+  }
+  // If the hours for that day are not available, set it as 'Closed'.
+  daysOfTheWeekFull.map(day => {
+    if (!Object.keys(hoursPerDayDict).includes(day)) {
+      hoursPerDayDict[day] = 'Closed';
+    }
+  });
+  return hoursPerDayDict;
+}
+
+// Takes in the index representing today's day of the week
+// and a dictionary mapping days to store hours. Returns a string
+// indicating when the store will next be open.
+function computeNextOpenDay(todaysDayIndex, hoursPerDayDict) {
+  // The store is closed, so we need the next day and its opening time
+  let nextDayIndex = (todaysDayIndex + 1) % 7;
+  let nextDay = daysOfTheWeekFull[nextDayIndex];
+  let nextDayStoreHours = hoursPerDayDict[nextDay];
+  // Loop until next open day is found.
+  // Case: All days are closed. We should limit to 7 loops.
+  for (let i = 0; i < 7; i++) {
+    if (nextDayStoreHours !== 'Closed') {
+      const nextDayHoursSplit = nextDayStoreHours.split('-');
+      const nextOpeningTime = nextDayHoursSplit[0]; // e.g. "8:30am"
+
+      return 'Closed until '
+        .concat(nextDay)
+        .concat(' ')
+        .concat(nextOpeningTime)
+        .replace('12am', 'Midnight');
```

Comment:
pretty minor but for code readability, it's better to use string interpolation:
```javascript
return `Closed until ${nextDay} ${nextOpeningTime}`.replace('12am', 'Midnight');
```

Reply:
good point ty fred!!